src/components/App.tsx

```diff
@@ -1277,6 +1281,20 @@ class App extends React.Component<ExcalidrawProps, AppState> {
     this.setState({ toastMessage: null });
   };
 
+  restoreFileFromShare = async () => {
+    const keys = await caches.keys();
+    const mediaCache = await caches.open(
+      keys.filter((key) => key.startsWith("media"))[0],
+    );
+
+    const file = await mediaCache.match("shared-file");
+    if (file) {
+      const blob = await file.blob();
+      this.loadFileToCanvas(blob);
+      await mediaCache.delete("shared-file");
```

Comment:
At this stage we also need to remove the URL parameter from the query. The History API can do this: https://stackoverflow.com/a/41061471.